presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5741,4 +5741,34 @@ public void testMapUnionSumOverflow()
                 "select y, map_union_sum(x) from (select 1 y, map(array['x', 'z', 'y'], cast(array[null,30, 32760] as array<smallint>)) x " +
                         "union all select 1 y, map(array['x', 'y'], cast(array[1,100] as array<smallint>))x) group by y", ".*Value 32860 exceeds MAX_SHORT.*");
     }
+
+    @Test
+    public void testLosslessJoin()
+    {
+        MaterializedResult countPart = computeActual("select partkey from part group by partkey");
+        MaterializedResult countLineitem = computeActual("select orderkey from lineitem group by orderkey");
+
+        MaterializedResult actual = computeActual("select p.partkey from part p left join lineitem l on p.partkey = l.orderkey group by p.partkey");
```

Comment:
Can we add a comment on what each test case is testing?

Reply:
done